packages/playwright-core/src/common/localUtils.ts

```diff
@@ -0,0 +1,247 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+
+import { assert, calculateSha1 } from '../utils';
+import { serializeClientSideCallMetadata } from '../utils';
+import { removeFolders } from '../utils/fileUtils';
+import { ManualPromise } from '../utils/manualPromise';
+import { fetchData } from '../utils/network';
+import { getUserAgent } from '../utils/userAgent';
+import { ZipFile } from '../utils/zipFile';
+import { yauzl, yazl } from '../zipBundle';
+import { HarBackend } from './harBackend';
+
+import type { Platform } from './platform';
+import type { Progress } from './progress';
+import type { HTTPRequestParams } from '../utils/network';
+import type * as channels from '@protocol/channels';
+import type * as har from '@trace/har';
+import type EventEmitter from 'events';
+import type http from 'http';
+
+
+export type StackSession = {
+  file: string;
+  writer: Promise<void>;
+  tmpDir: string | undefined;
+  callStacks: channels.ClientSideCallMetadata[];
+};
+
+export async function zip(platform: Platform, stackSessions: Map<string, StackSession>, params: channels.LocalUtilsZipParams): Promise<void> {
```

Comment:
I wonder why export a bunch of free functions instead of a class or multiple classes with `platform`, `stackSessions` and `harBackends` encapsulated?